src/compiler/types.ts

```diff
@@ -603,15 +603,75 @@ namespace ts {
 
     export type DeclarationName = Identifier | StringLiteral | NumericLiteral | ComputedPropertyName | BindingPattern;
 
-    export interface RealDeclaration extends Node {
+    export interface DeclarationBase extends Node {
         _declarationBrand: any;
         name?: DeclarationName;
     }
 
-    // Binary expressions can be declarations if they are 'exports.foo = bar' expressions in JS files
-    export type Declaration = RealDeclaration | BinaryExpression;
-
-    export interface DeclarationStatement extends RealDeclaration, Statement {
+    export type Declaration =

```

Comment:
Based on a couple of before/after runs just timed using `time`, the time goes from `46.1` to `47.7` seconds, or 3.3% slower. I think that's OK &mdash; @mhegazy do you have an opinion?